### DISCUSSION TITLE
How to make custom extension methods visible to client

### DISCUSSION THREAD

**Discussion by jason-downs** [serverpod/serverpod#2012] [Q&A] [answered]
In the Serverpod docs, the section "Custom Methods" says you can add methods to a class that's definited in the protocol yaml files by using Dart's extension feature.

https://docs.serverpod.dev/next/concepts/models#custom-methods

`Sometimes you will want to add custom methods to the generated classes. The easiest way to do this is with [Dart's extension feature](https://dart.dev/language/extension-methods).`

However, this makes it accessible only to the server-side code. In many cases it is useful to have these custom methods avaiable to the client flutter app as well. I would imagine that adding an `export` statement to the server-side `src/generated/protocol.dart` will do the trick but we should in principle avoid modifying this auto-generated code.

Is there an intended way to do this - or if not then I think it should be a feature to add in a future serverpod version.

(Note this is similar to the discussion in #1571 but in that case it's about extending one of serverpod's built-in classes whereas here I'm interested in extending the custom yaml-defined classes.)

**Comment by SandPod** [comment, marked as answer]
Hey @jason-downs 👋 

I think this question was answered here: https://github.com/serverpod/serverpod/discussions/1716

TLDR: Sharing code on models between client and server is complex currently but we want to make is simpler.
